contracts/extensions/CoinMachine.sol

```diff
@@ -0,0 +1,197 @@
+/*
+  This file is part of The Colony Network.
+
+  The Colony Network is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  The Colony Network is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with The Colony Network. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+pragma solidity 0.5.8;
+pragma experimental ABIEncoderV2;
+
+import "./../../lib/dappsys/math.sol";
+import "./../colony/IColony.sol";
+import "./../common/ERC20Extended.sol";
+
+// ignore-file-swc-108
+
+
+contract CoinMachine is DSMath {
+
+  IColony colony;
+
+  constructor(address _colony) public {
+    colony = IColony(_colony);
+  }
+
+  // Storage
+  address purchaseToken; // The token in which we receive payments, 0x0 for eth
+
+  uint256 periodLength; // Duration of a sale period (i.e. one hour)
+  uint256 windowSize; // Number of periods in the price averaging window (i.e. 10)
+
+  uint256 targetPerPeriod; // Target number of tokens to sell in a period
+  uint256 maxPerPeriod; // Maximum number of tokens sellable in a period
+  int256 tokenSurplus; // Total tokens undersold (i.e. below target), negative means deficit
```

Comment:
Why are we spending gas to track this on-chain? Perhaps I'm being blind, but I can't see how it's used.

Reply:
It's not, it's just a convenience. I suppose we could remove it.